src/java.base/share/classes/java/lang/ref/Reference.java

```diff
@@ -511,29 +544,35 @@ protected Object clone() throws CloneNotSupportedException {
     }
 
     /**
-     * Ensures that the object referenced by the given reference remains
-     * <a href="package-summary.html#reachability"><em>strongly reachable</em></a>,
-     * regardless of any prior actions of the program that might otherwise cause
-     * the object to become unreachable; thus, the referenced object is not
+     * Ensures that the given object remains
+     * <a href="package-summary.html#reachability"><em>strongly reachable</em></a>.
+     * This reachability is assured regardless of any optimizing transformations
+     * the VM may perform that might otherwise allow the object to become
```

Comment:
since "the virtual machine" is spelt out later in the documentation I think we should do the same here.

```suggestion
     * the virtual machine may perform that might otherwise allow the object to become
```